sys/linux/damon.txt

```diff
@@ -0,0 +1,97 @@
+# Copyright 2018 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+include <uapi/linux/fcntl.h>
+include <uapi/linux/mount.h>
+
+resource fd_damon_target_ids[fd]
+resource fd_damon_attrs[fd]
+resource fd_damon_monitor_on[fd]
+resource fd_damon_init_regions[fd]
+resource fd_damon_schemes[fd]
+resource fd_damon_kdamond_pid[fd]
+resource fd_damon_mk_contexts[fd]
+resource fd_damon_rm_contexts[fd]
+
+openat$damon_target_ids(fd const[AT_FDCWD], file ptr[in, string["/sys/kernel/debug/damon/target_ids"]], flags flags[open_flags], mode flags[open_mode]) fd_damon_target_ids
+openat$damon_attrs(fd const[AT_FDCWD], file ptr[in, string["/sys/kernel/debug/damon/attrs"]], flags flags[open_flags], mode flags[open_mode]) fd_damon_attrs
+openat$damon_init_regions(fd const[AT_FDCWD], file ptr[in, string["/sys/kernel/debug/damon/init_regions"]], flags flags[open_flags], mode flags[open_mode]) fd_damon_init_regions
+openat$damon_monitor_on(fd const[AT_FDCWD], file ptr[in, string["/sys/kernel/debug/damon/monitor_on"]], flags flags[open_flags], mode flags[open_mode]) fd_damon_monitor_on
+openat$damon_schemes(fd const[AT_FDCWD], file ptr[in, string["/sys/kernel/debug/damon/schemes"]], flags flags[open_flags], mode flags[open_mode]) fd_damon_schemes
+openat$damon_kdamond_pid(fd const[AT_FDCWD], file ptr[in, string["/sys/kernel/debug/damon/kdamond_pid"]], flags flags[open_flags], mode flags[open_mode]) fd_damon_kdamond_pid
+openat$damon_mk_contexts(fd const[AT_FDCWD], file ptr[in, string["/sys/kernel/debug/damon/mk_contexts"]], flags flags[open_flags], mode flags[open_mode]) fd_damon_mk_contexts
+openat$damon_rm_contexts(fd const[AT_FDCWD], file ptr[in, string["/sys/kernel/debug/damon/rm_contexts"]], flags flags[open_flags], mode flags[open_mode]) fd_damon_rm_contexts
+
+write$damon_target_ids(fd fd_damon_target_ids, buf ptr[in, damon_target_id_input], count len[buf])
+write$damon_attrs(fd fd_damon_attrs, buf ptr[in, damon_attrs], count len[buf])
+write$damon_init_regions(fd fd_damon_init_regions, buf ptr[in, damon_init_regions], count len[buf])
+write$damon_monitor_on(fd fd_damon_monitor_on, buf ptr[in, string["on"]], count len[buf])
+write$damon_schemes(fd fd_damon_schemes, buf ptr[in, damon_schemes], count len[buf])
+write$damon_mk_contexts(fd fd_damon_mk_contexts, buf ptr[in, damon_contexts], count len[buf])
+write$damon_rm_contexts(fd fd_damon_rm_contexts, buf ptr[in, damon_contexts], count len[buf])
+
+read$damon_target_ids(fd fd_damon_target_ids, buf ptr[out, array[int8]], count len[buf])
```

Comment:
Generic read and close are already defined for `fd` resource, since `fd_damon_*` resources are inherited from `fd` it's not necessary to define read and close. Remove them, the fuzzer will invoke read and close on these fd's anyway.
It's only necessary if we need to pass some new types of data to syscalls (like in the case of write syscalls).